src/main/groovy/edu/oregonstate/mist/trailsapi/resources/TrailResources.groovy

```diff
@@ -81,4 +81,25 @@ public class TrailsResource extends Resource {
 	Boolean trailValidator(Trail trail) {
 		trail.name && trail.zipCode && trail.difficulty
 	}
+
+	@GET
+	@Produces(MediaType.APPLICATION_JSON)
+	Response getTrail (@QueryParam("name") Optional<String> name,
+					@QueryParam("difficulty") Optional<String> difficulty,
+					@QueryParam("mostDifficult") Optional<String> mostDifficult,
+					@QueryParam("leastDifficult") Optional<String> leastDifficult,
+					@QueryParam("zipCode") Optional<Integer> zipCode,
+					@QueryParam("smallDrop") Optional<Boolean> smallDrop,
+					@QueryParam("largeDrop") Optional<Boolean> largeDrop,
+					@QueryParam("woodRide") Optional<Boolean> woodRide,
+					@QueryParam("skinny") Optional<Boolean> skinny,
+					@QueryParam("largeJump") Optional<Boolean> largeJump,
+					@QueryParam("smallJump") Optional<Boolean> smallJump,
+					@QueryParam("gap") Optional<Boolean> gap) {
+			List<Trail> trails = trailDAO.getTrailByQuery(name.or(""), difficulty.or(""),
```

Comment:
A lot of your query parameters are just the names of the fields in your `Trail` class. One idea is to change the DAO method to accept a trail object (as `@BindBean`) so that you don't have to pass all these parameters separately.

Also, do we need to have `Optional<>` for all these parameters? If the parameters are null, your DAO method will treat them as null.

Reply:
A few notes on using BindBean:
* The most and least difficult fields are not fields in the Trail object. They could be passed separately along with a trail object, but
* BindBean fails when fields in the object are null Booleans. It only fails with the Boolean data type, and it only applies if it is a field inside of an object that is being bound by BindBean. Binding null Booleans individually is fine.

I am pretty sure it is a result of this bug, which appears to be fixed in JDBC v3 but not v2: https://github.com/jdbi/jdbi/issues/97

Also, Optional<> definitely isn't necessary, so I'll remove those.

Reply:
Thanks for looking into that.